### PR TITLE
Add Mac OSX support

### PR DESCRIPTION
**Pull request by johnwason**
This PR contains patches and a GitHub build action for Mac OSX.

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@johnwason Let me know when this is ready to review. I think it would be good to leverage the colcon-build action and update it to support MacOS.

**Comment by johnwason** [comment]
@Levi-Armstrong sure, sounds good. I am getting lots of errors on CI that I am not getting locally so it may take some time to figure out.

**Comment by johnwason** [comment]
The conda build failure is unrelated to this PR. This is ready to merge.

File: tesseract_common/include/tesseract_common/class_loader.hpp [hunk 1]
```diff
@@ -186,7 +186,7 @@ std::vector<std::string> ClassLoader::getAvailableSections(const std::string& li
     if (include_hidden)
       return false;
 
-    return (section.substr(0, 1) == ".");
+    return (section.substr(0, 1) == ".") || (section.substr(0, 1) == "_");
```
Comment (Levi-Armstrong) [inline tesseract_common/include/tesseract_common/class_loader.hpp:189]: Why was this change needed?

File: tesseract_common/test/plugin_loader_unit.cpp [hunk 2]
```diff
@@ -80,7 +80,7 @@ TEST(TesseractClassLoaderUnit, LoadTestPlugin)  // NOLINT
   }
 
 // For some reason on Ubuntu 18.04 it does not search the current directory when only the library name is provided
-#if BOOST_VERSION > 106800
+#if BOOST_VERSION > 106800 && !__APPLE__
```
Comment (Levi-Armstrong) [inline tesseract_common/test/plugin_loader_unit.cpp:83]: Why does this not work on MacOS?

File: tesseract_common/include/tesseract_common/class_loader.hpp (hunk 1 above)
Comment (johnwason) [reply tesseract_common/include/tesseract_common/class_loader.hpp:189]: Private symbols appear to start with underscores on macos

File: tesseract_common/test/plugin_loader_unit.cpp (hunk 2 above)
Comment (johnwason) [reply tesseract_common/test/plugin_loader_unit.cpp:83]: Not sure but the other loader tests are working

**Review by Levi-Armstrong — APPROVED** [review]
I will squash merge when CI finishes.
